docs/APIReference-Container.md

```diff
@@ -383,9 +383,11 @@ module.exports = Relay.createContainer(Feed, {
 });
 ```
 
-> Note
+> Notes
 >
 > `setVariables` does not immediately mutate `variables`, but creates a  pending state transition. `variables` will continue returning the previous values until `this.props` has been populated with data that fulfills the new variable values.
+>
+> If `setVariables` is used to change `first` and `hasNextPage` is `false` (also `last` and `hasPreviousPage`), Relay does not send a request to the server because there isn't any more data as far as Relay is concerned.
```

Comment:
> because there isn't any more data as far as Relay is concerned.

It's not really Relay that is making this choice; it is the schema. The contract for `hasNextPage` is that it should be true if there are more edges to fetch, and false if there are no more edges. So it might be more accurate to rephrase this as "because the server has indicated there are no more edges that can be fetched."